cirq/sim/simulator.py

```diff
@@ -114,6 +114,72 @@ def _run(
         """
         raise NotImplementedError()
 
+    def compute_samples_displays(
+            self,
+            program: Union[circuits.Circuit, schedules.Schedule],
+            param_resolver: Optional[study.ParamResolver] = None,
+    ) -> study.ComputeDisplaysResult:
+        """Computes SamplesDisplays in the supplied Circuit or Schedule.
+
+        Args:
+            program: The circuit or schedule to simulate.
+            param_resolver: Parameters to run with the program.
+
+        Returns:
+            ComputeDisplaysResult for the simulation.
+        """
+        return self.compute_samples_displays_sweep(
+            program, [param_resolver or study.ParamResolver({})])[0]
+
+    def compute_samples_displays_sweep(
+            self,
+            program: Union[circuits.Circuit, schedules.Schedule],
+            params: Optional[study.Sweepable] = None
+    ) -> List[study.ComputeDisplaysResult]:
+        """Computes SamplesDisplays in the supplied Circuit or Schedule.
+
+        In contrast to `compute_displays`, this allows for sweeping
+        over different parameter values.
+
+        Args:
+            program: The circuit or schedule to simulate.
+            params: Parameters to run with the program.
+
+        Returns:
+            List of ComputeDisplaysResults for this run, one for each
+            possible parameter resolver.
+        """
+        circuit = (program if isinstance(program, circuits.Circuit)
+                   else program.to_circuit())
+        param_resolvers = study.to_resolvers(params or study.ParamResolver({}))
+
+        compute_displays_results = []  # type: List[study.ComputeDisplaysResult]
+        for param_resolver in param_resolvers:
+            display_values = {}  # type: ignore
+            for i, moment in enumerate(circuit):
+                displays = (op for op in moment
+                            if isinstance(op, ops.SamplesDisplay))
+                preceding_circuit = circuit[:i]
+                for display in displays:
+                    measurement_key = str(display.key)
+                    measurement_circuit = circuits.Circuit.from_ops(
+                        display.measurement_basis_change(),
+                        ops.measure(*display.qubits,
+                                    key=measurement_key)
+                    )
+                    measurements = self._run(
```

Comment:
This looks like it is scaling like O(moments^2)?

Reply:
This scaling seems unavoidable to me with the current interface because of the need to generate new circuits to submit to `_run`. Do you see a good way around it?

Reply:
I agree that it's unavoidable assuming you only have access to a sampling method.